### PR TITLE
feat(user): create user schema

### PR DESCRIPTION
**Pull request by guhur**
This PR offers a new schema: user.

Users were thought first as an abstraction for any organizer. When a poll is created without any `user-id`, the back should create a dummy user.

But actually, we can extend the schema to any participant. In this case, we should clarify how to create a poll with optional participant.

### CODE REVIEW

File: openapi/paths/users.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        update-poll:
+          $ref: ../components/links/update-poll.yaml
+        delete-poll:
+          $ref: ../components/links/delete-poll.yaml
+        create-ballot:
+          $ref: ../components/links/create-ballot.yaml
+        list-poll-results:
+          $ref: ../components/links/list-poll-results.yaml
+    '400':
+      description: Bad Request. Expect at least 2 proposals and a title.
```
[the remark this answers is not in the document]
Comment (domi41) [reply openapi/paths/users.yaml:45]: https://forum.mieuxvoter.fr/t/terminologie-mieuxvoter/42

**Comment by guhur** [comment]
To answer one of your comments we haven't thought long about the dummy user. 

What would you suggest instead? I understand you want to use NULL. Would it work as well with participants?

**Comment by domi41** [comment]
The advantage of dummy user creation is accountability, since users already handle the IP address logic, usually.
Setting a NULL would require storing the IP of the author, in case they are abusers.

---

Regarding participants, it's another story.  If we want people to be able to judge anonymously (and perhaps multiple times), sure we can have NULL values ; we probably need to store the IP somewhere, for accountability.  We can also use ballot tokens for the organizer to distribute, in order to allow anonymous voting while preventing brigading|spamming that would skew results.

File: openapi/paths/polls.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 get:
-  operationId: get-polls
-  summary: Gets all polls.
+  operationId: get-users
+  summary: Gets all users.
```
[the remark this answers is not in the document]
Comment (guhur) [reply openapi/paths/polls.yaml:3]: Right that sounds like a mistake thanks

File: mv-openapi.yaml [hunk 3]
```diff
@@ -264,7 +355,28 @@ components:
...
+          example: 0
+          readOnly: true
+        name:
+          type: string
+          description: Unique user-name
+          example: john.doe
+        description:
+          type: string
+          description: An optional description of the user
+          example: >-
```
[the remark this answers is not in the document]
Comment (guhur) [reply mv-openapi.yaml:377]: Yes

File: mv-openapi.yaml [hunk 4]
```diff
@@ -310,6 +426,10 @@ components:
             Unique but short name of the mention, like "Excellent" or "To
             reject".
           example: Excellent
+        description:
+          type: string
+          description: An optional description of the mention
+          example: 'In this poll, the mention "To reject" disqualifies the proposal.'
```
[the remark this answers is not in the document]
Comment (guhur) [reply mv-openapi.yaml:432]: OK

**Comment by guhur** [comment]
I think it is a good idea to provide a NULL token for both organizers and participants. I don't like it is necessarily the responsibility of the front to create them. The back can do that.

**Comment by guhur** [comment]
In OpenAPI, parameters are by default optional. So I created a required field for the other, to show that the user-id is optional.

**Review by domi41 — APPROVED** [review]

**Comment by domi41** [comment]
Should we merge this so I can rebase my review on it?

@guhur

**Pull request merged**